file.go

```diff
@@ -76,6 +79,67 @@ func (c *Cache) SaveToFileConcurrent(filePath string, concurrency int) error {
 	return nil
 }
 
+// SaveToFileConcurrentWithTimeout basically does the same things which SaveToFileConcurrent does.
+//
+// The difference is that SaveToFileConcurrentWithTimeout accepts timeout parameter to
+// limit the time for saving the data when cached data is too huge.
+//
+// The saved data may be loaded with LoadFromFile*.
+//
+// See also SaveToFileConcurrent.
+func (c *Cache) SaveToFileConcurrentWithTimeout(filePath string, concurrency int, timeout time.Duration) error {
```

Comment:
I suppose that passing `io.Writer` instead of filepath (and opening file inside) will be much more flexible and easier to test. WDYT?

Reply:
@cristaloleg Thanks for the comment but I'm sorry that I don't quite understand your point "passing io.Writer and opening the file inside" 😭 Could you please elaborate? In this change, I wanted to make the function signature look like existing `SaveToFileConcurrent`.

Reply:
I was thinking about slightly another API:

`func (c *Cache) SaveConcurrentWithTimeout(w io.Writer, concurrency int, timeout time.Duration) error {`

Anyway, I'm the author of fastcache, someone else might have another view on this.